### PR TITLE
Added code for user crud operation using mongo query

### PR DESCRIPTION
**Pull request by Manish21-v**

### CODE REVIEW

File: MongoUsingMongoose/controller.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+              message: "User not found with email " + req.params.email
+          });                
+      }
+      return res.status(500).send({
+          message: "Error retrieving user with email " + req.params.email
+      });
+  });
+}
+
+exports.autheticateToken = (req,res,next) =>{
```
Comment (sunil4sonawane) [inline MongoUsingMongoose/controller.js]: Correct the spelling for authenticateToken

File: MongoUsingMongoose/controller.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    const authHeader = req.headers['authorization'];
+    const token =authHeader && authHeader.split(' ')[1];
+    if(token == null) return res.status(401).send({
+        message : "Token not provided"
+    });
+    console.log(token);
```
Comment (sunil4sonawane) [inline MongoUsingMongoose/controller.js]: remove console.log statements

File: MongoUsingMongoose/controller.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  .then(user => {
+      if(!user) {
+          return res.status(404).send({
+              message: "User not found with email " + req.params.email
+          });            
+      }  
+      if((req.body.email == user.email) && (req.body.password == user.password))
+      {
+          const username = {name : user.name};
+          const accessToken = jwt.sign(username,process.env.ACCESSTOKEN,{expiresIn : '40s'});
```
Comment (sunil4sonawane) [inline MongoUsingMongoose/controller.js]: Token should be generated and sent back to user. Not from ENV

File: MongoUsingMongoose/key.pem [hunk 4]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (sunil4sonawane) [inline MongoUsingMongoose/key.pem]: WE should not push this file to git

File: MongoUsingMongoose/sendEmail.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+//function to send the email
+function emailData(mail_Obj) {
+  let transporter = nodeMailer.createTransport({
+    host: 'smtp.gmail.com',
+    port: 465,
+    secure: true,
+    auth: {
+      //assigning the email id and password of the sender
+        user: 'vitkarmanish21@gmail.com',
+        pass: 'zuykdecyecslslgu'
```
Comment (sunil4sonawane) [inline MongoUsingMongoose/sendEmail.js]: should be picked from ENV

File: MongoUsingMongoose/server.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+var httpServer = http.createServer(app);
+// parse requests of content-type 
+app.use(bodyParser.urlencoded({ extended: true }))
+// parse requests of content-type - application/json
+app.use(bodyParser.json())
+//Reference of the router file
+const routerObj = require('./routes');
+app.use('/',routerObj);
+//connecting to the database !!!
+dbConfig.connect();
```
Comment (sunil4sonawane) [inline MongoUsingMongoose/server.js]: server should not start if db connection is failed

**Review by sunil4sonawane — COMMENTED** [review]
Please check comments

**Review by sunil4sonawane — CHANGES_REQUESTED** [review]
Please check comments

File: MongoUsingMongoose/controller.js (hunk 2 above)
Comment (Manish21-v) [reply MongoUsingMongoose/controller.js]: removed

File: MongoUsingMongoose/controller.js (hunk 3 above)
Comment (Manish21-v) [reply MongoUsingMongoose/controller.js]: changed the name of the variable that was creating confusion

File: MongoUsingMongoose/key.pem (hunk 4 above)
Comment (Manish21-v) [reply MongoUsingMongoose/key.pem]: removed this file

File: MongoUsingMongoose/sendEmail.js (hunk 5 above)
Comment (Manish21-v) [reply MongoUsingMongoose/sendEmail.js]: Done

File: MongoUsingMongoose/server.js (hunk 6 above)
Comment (Manish21-v) [reply MongoUsingMongoose/server.js]: Done

**Review by sunil4sonawane — APPROVED** [review]

**Pull request merged**